### DISCUSSION TITLE
Fundamental differences between Nue and Marko?

### DISCUSSION THREAD

**Discussion by batonac** [nuejs/nue#105] [Q&A]
Hello 👋

I'm intrigued by this project, but on the face of it Nue seems to adopt a very similar approach vis-a-vis [Marko](https://github.com/marko-js/marko).

Anyone care to explain the fundamental differences in approach that prompted the creation of a new framework?

Thanks

**Comment by tipiirai** [comment]
Hey,

Indeed. I can see a lot of similarities there. Most importantly, Nue and Marko are HTML-based, the scripting is class-based, and the messaging on the website feels somewhat similar. There are, of course, tons of differences in the templating syntax when you start digging deeper, particularly on how components are defined. Nue attempts to stay as far away from JavaScript and TypeScript vocabulary as possible. Nue is also strongly against CSS-in-JS and embraces semantic HTML and external stylesheets — the way CSS is intended to be used. Overall the goal is to stay as close to standards as possible. We need a reactive library like Marko since web components are super limited and a bit of a mess to work with. 

The biggest difference, however, is in the grand vision. Nue attempts to be an alternative to builders like Next.js, and Astro — and the Marko-like Nue JS library is just a part of the story. The goal is to reveal this vision early next month after which, the difference between Marko and everyone else will be super clear.

**Comment by batonac** [comment]
Thanks for the comments and projections. I look forward to seeing more about how this develops in the future.

**Reply by tipiirai** [reply]
Here's the latest update: https://nuejs.org/blog/nuekit-010/
